### ISSUE TITLE
GhcModType error w/ ghc-mod 5.5.0

### ISSUE THREAD

**Issue by jonschoning** [eagletmt/ghcmod-vim#94]
### GhcModType
```
ghcmod#command#type: Cannot guess type
```
### versions
```
ghcmod-vim sha: a7c76b9
ghcmod.vim version: 1.3.1
vimproc.vim: 902
ghc-mod version: 5.5.0
ghc version: 7.10.3
```

### strace
```
    read(18, "Not a number \"Main\"\n\nUsage: ghc-mod type FILE LINE COL\n  Get the type of the expression under (LINE,COL)\n", 65523) = 105
```

### ~/.vim/bundle/ghcmod-vim/test.sh
```
Running basedir
Total 6, Success 6, Failure 0
Running build_command
Total 4, Success 4, Failure 0
Running check
Total 5, Success 5, Failure 0
Running command_check
Total 4, Success 4, Failure 0
Running command_sig_codegen
test_command_sig_function failed at [function 1..2..19, line 2]
expected to match ^func x y f =   actual: ''
test_command_sig_instance failed at [function 1..3..19, line 2]
expected to match ^\s*cInt x =   actual: ''
Total 2, Success 0, Failure 2

"test/data/sig/SigFunction.hs" 
"test/data/sig/SigFunction.hs" 3L, 70C
Cannot generate initial code
"test/data/sig/SigInstance.hs" 
"test/data/sig/SigInstance.hs" 9L, 138C
Cannot generate initial code
Running command_split
test_command_split failed at [function 1..2..10, line 2]
expected 'f x = undefined' but got 'f [] = undefined'
Total 1, Success 0, Failure 1

"test/data/split/Split.hs" 
"test/data/split/Split.hs" 4L, 54C
No splittable constructor
Running command_type
test_command_type_toggle raised at [function 1..4..<SNR>9_type, line 2]
Vim(return):E121: Undefined variable: b:ghcmod_type
test_command_type_match_change raised at [function 1..6, line 4]
Vim(let):E121: Undefined variable: b:ghcmod_type
test_command_type raised at [function 1..3..<SNR>9_type, line 2]
Vim(return):E121: Undefined variable: b:ghcmod_type
test_command_type_highlight raised at [function 1..5..<SNR>9_type, line 2]
Vim(return):E121: Undefined variable: b:ghcmod_type
test_command_type_clear failed at [function 1..7..20, line 2]
expected to exist: b:ghcmod_type
Total 5, Success 0, Failure 5

"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
ghcmod#command#type: Cannot guess type
"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
ghcmod#command#type: Cannot guess type
"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
ghcmod#command#type: Cannot guess type
"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
ghcmod#command#type: Cannot guess type
"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
ghcmod#command#type: Cannot guess type
Running detect_module
Total 4, Success 4, Failure 0
Running expand
test_expand_whitespace failed at [function 1..4..15, line 7]
expected to satisfy any: [{'text': 'Invalid option `-b '''}, {'text': ''''''}, {'text': ''}, {'text': 'Usage: ghc-mod expand FILES..'}, {'text': 'Like `check'' but also pass `-ddump-splices'' to GHC'}]
test_expand failed at [function 1..3..15, line 7]
expected to satisfy any: [{'text': 'Invalid option `-b '''}, {'text': ''''''}, {'text': ''}, {'text': 'Usage: ghc-mod expand FILES..'}, {'text': 'Like `check'' but also pass `-ddump-splices'' to GHC'}]
Total 2, Success 0, Failure 2

"test/data/th with whitespace/Fuga.hs" 
"test/data/th with whitespace/Fuga.hs" 13L, 192C
"test/data/th/Fuga.hs" 
"test/data/th/Fuga.hs" 13L, 192C
Running info
test_info failed at [function 1..3..19, line 2]
expected to match ^bar :: \[Char\]  actual: 'Cannot show infoEXCEPTION: info:     Not in scope: ‘Foo’'
test_info_syntax_error failed at [function 1..4..19, line 2]
expected to match ^Cannot show info  actual: 'EXCEPTION: info:               parse error (possibly incorrect indentation or mismatched brackets)Cannot show info'
Total 2, Success 0, Failure 2

"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
"test/data/failure/SyntaxError.hs" 
"test/data/failure/SyntaxError.hs" 5L, 52C
Running lint
Total 5, Success 5, Failure 0
Running split
test_split failed at [function 1..2..9, line 2]
expected ['f [] = undefined', 'f (x:xs) = undefined'] but got []
Total 1, Success 0, Failure 1

"test/data/split/Split.hs" 
"test/data/split/Split.hs" 4L, 54C
Running type
test_type failed at [function 1..3..11, line 2]
expected [[[4, 7, 4, 10], '[Char]'], [[4, 7, 4, 16], '[Char]'], [[4, 1, 4, 16], '[Char]']] but got []
Total 2, Success 1, Failure 1

"test/data/with-cabal/src/Foo.hs" 
"test/data/with-cabal/src/Foo.hs" 4L, 49C
"test/data/failure/Main.hs" 
"test/data/failure/Main.hs" 5L, 62C

```

**Comment by DanielG** [comment]
Should be fixed once https://github.com/eagletmt/ghcmod-vim/pull/93 is merged.
